website/static/js/filters.js

```diff
@@ -0,0 +1,149 @@
+                        if (match.indexOf(el.querySelector(selector).checked) === -1) {
+                            return false;
+                        }
+                    }
+                }
+            }
+            return true;
+        };
+
+        var toggle = function() {
+            var activeItems = $(itemsSelector);
+            $(itemsSelector).each(function() {
+                var self = this;
+                if (!search(self) || !filter(self)) {
+                    activeItems.splice(activeItems.index(self), 1);
+                    $(self).fadeOut();
+                }
+            });
+            $(activeItems).fadeIn();
+            if (settings.callback !== undefined) {
+                var empty = [];
+                var filtered = [];
+                for (var i = 0, selector; selector = settings.items[i]; i ++) {
+                    var active = $(activeItems).filter(selector);
+                    var items = $(selector);
+                    if (active.length < items.length) {
+                        filtered.push(selector);
+                    }
+                    if (active.length === 0) {
+                        empty.push(selector);
+                    }
+                }
+                settings.callback(filtered, empty);
+            }
+        };
+
+        var clickFunction = function() {
+            var self = this;
+            $(self).toggleClass('btn-primary btn-default');
+            var group = self.parentElement.parentElement.id;
+            var match = settings.groups[group].buttons[self.id];
+            try {
+                var index = filterConstraints[group].indexOf(match);
+                if (index === -1) {
+                    filterConstraints[group].push(match);
+                }
+                else {
+                    filterConstraints[group].splice(index, 1);
+                    if (filterConstraints[group].length === 0) {
+                        delete filterConstraints[group];
+                    }
+                }
+            }
+            catch(TypeError) {
+                filterConstraints[group] = [match];
+            }
+            toggle();
+        };
+
+        for (var group in settings.groups) {
+            if (settings.groups.hasOwnProperty(group)) {
+                for (var value in settings.groups[group].buttons) {
+                    if (settings.groups[group].buttons.hasOwnProperty(value)) {
+                        $('#' + value).on('click', clickFunction);
+                    }
+                }
+            }
+        }
+
+        function keyupFunction() {
+            var self = this;
+            var selector = settings.inputs[self.id];
+            searchConstraints[selector] = $(self).val().toLowerCase();
+            toggle();
+        }
+
+        for (var key in settings.inputs) {
+            if (settings.inputs.hasOwnProperty(key)) {
+                $('#' + key).keyup(keyupFunction);
```

Comment:
Where/when would a user use `keyUp` on the contributor page (besides scrolling)? I'm not sure I understand what this is for.